In [1]:
#hide
%load_ext autoreload
%autoreload 2

# Dimension Reduction

> The function performs dimensionality reduction, pre-processing the data and comparing the reconstruction error via PCA and autoencoder.

Reference: 
- https://towardsdatascience.com/dimensionality-reduction-pca-versus-autoencoders-338fcaf3297d
- https://towardsdatascience.com/autoencoders-vs-pca-when-to-use-which-73de063f5d7

## Install

`pip install embedding-tools`

## How to use

In [2]:
#hide
import pickle
import pandas as pd

with open('testing_data.pickle', 'rb') as handle:
    testing_data = pickle.load(handle)
    
print ("Data's size: ", testing_data.shape)
print ("Dimension:   ", testing_data.shape[1])

Data's size:  (863, 768)
Dimension:    768


In [3]:
from short_text_analyzer.core import *

**Input data:**
The input matrix has a size of 863 $\times$ 768.

In [4]:
print ("Data's size: ", testing_data.shape)
print ("Dimension:   ", testing_data.shape[1])

Data's size:  (863, 768)
Dimension:    768


**Performing dimension reduction:** we will reduce the number of dimension from 768 to 2. 

In [5]:
dim_reducer = dimensionReducer(analyzer.embeddingRaw, 2, 0.002)
dim_reducer.fit()

NameError: name 'dimensionReducer' is not defined

**Calculating the MSE of the reconstructed vectors**

In [ ]:
dim_reducer.rmse_result

***

In [7]:
#hide
from nbdev.export import notebook2script; notebook2script()

Converted 00_dimensionReducer.ipynb.
Converted index.ipynb.
